packages/amplify-provider-awscloudformation/src/aws-utils/aws-ecr.ts

```diff
@@ -0,0 +1,26 @@
+import type AWS from "aws-sdk";
+import aws from './aws';
+import configurationManager from "../configuration-manager";
+
+class ECR {
+  public ecr: AWS.ECR;
+
+  constructor(private readonly context: any, options = {}) {
```

Comment:
Async constructor is an anti pattern we are trying to get away from. Can you use `getInstance` similar to we do with s3.
https://github.com/aws-amplify/amplify-cli/blob/29154ce4e07f6307d0f350edb2e03420bd72dbbb/packages%2Famplify-provider-awscloudformation%2Fsrc%2Faws-utils%2Faws-s3.ts#L27-L39